src/passes/strings-encoding/StringEncoding.cpp

```diff
@@ -370,6 +370,7 @@ bool StringEncoding::injectOnStackLoop(BasicBlock& BB, Instruction& I, Use& Op,
   }
   SmallVector<ReturnInst*, 8> Returns;
   CloneFunctionInto(NewF, FDecode, VMap, CloneFunctionChangeType::DifferentModule, Returns);
+  NewF->setDSOLocal(true);
```

Comment:
Is this relevant outside dynamic shared objects? And how is it related to the other two commits?

Reply:
There existed an issue with the current linkage of `__omvll_decode`, as we happened to trigger [this check](https://github.com/llvm/llvm-project/blob/da90fd75241d5b149fd307e3eeae84f7317d9d9b/llvm/lib/IR/Verifier.cpp#L745-L747) ([CI job failure](https://github.com/antoniofrighetto/o-mvll/actions/runs/6882743561/job/18721878292)). I suspect that the specifier gets dropped in `CloneFunctionInto` because of the change type. This is independent of the patch, and I assume this was not caught before due to missing module verification.